### ISSUE TITLE
Empty window opens with internalConsole during debug

### ISSUE THREAD

**Issue by csorvagep** [microsoft/vscode-cpptools#7528]
Bug type: Debugger

**Describe the bug**
- OS and Version: Windows 10 20H2
- VS Code Version: 1.56.2
- C/C++ Extension Version: 1.4.0-insiders

After setting the launch.json as the following, an external window is opened at debug. All output appears in the debug console, but an empty console window is there.

``` json
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug (win)",
            "type": "cppvsdbg",
            "request": "launch",
            "program": "${command:cmake.launchTargetPath}",
            "args": [ ],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "console": "internalConsole",
            "internalConsoleOptions": "openOnSessionStart"
        }
```

I looked up the documentation for the launch.json, but there is no mention of "console" option, only the old "externalConsole" option is there, which is deprecated.

**Comment by sean-mcmanus** [comment]
"console" got added recently. The documentation hasn't gotten updated yet.

**Comment by csorvagep** [comment]
Yes I thought so, since this in on insiders. I've only wrote that, to ensure that I've looked it up and creating an issue is not my first thing. :)

**Comment by WardenGnaw** [comment]
Documentation is currently in the intellisense of the `launch.json`.

![image](https://user-images.githubusercontent.com/3953714/118533671-3198a300-b6fd-11eb-812c-512210992900.png)
![image](https://user-images.githubusercontent.com/3953714/118533715-3f4e2880-b6fd-11eb-9ed0-b35c7401be42.png)

The behavior you are seeing seems to be set to `newExternalWindow`, but I am unable to recreate the new window showing up with just `internalConsole`

**Comment by csorvagep** [comment]
Yes, the window is similar to a `newExternalWindow` (so it's not an `externalTerminal` window) but all output appears in the debug console.

I've created a simple test executable project with cmaketools quick start, then placed a breakpoint on the "Hello, world!" line. Created a windows launch configuration, set the program to: `${command:cmake.launchTargetPath}` and the console to `internalConsole`. After starting a debugging session an empty window opens and the focus is on that window. (that is the annoying part). After continuing the program the text appears in the debug console, and the window closes. 

Should I copy some logs or else in order to reproduce?

**Comment by WardenGnaw** [comment]
Can you share the simple test project, steps to build and the `launch.json` if its different from the one above?

**Comment by csorvagep** [comment]
I'm using the CMakeTools extension for the project generation and build:
- Command palette: CMake: Quick Start, set project name, executable target
- Kit selected: Visual Studio Community 2019 - amd64
- Configure and build (or F7)
- Place breakpoint on line 4
- Run and Debug panel create a launch.json file link, C++ (Windows)
- Change program in launch.json to `${command:cmake.launchTargetPath}` and console to `internalConsole`
- Debug the program

Here are the generated files:
main.cpp:

```cpp
#include <iostream>

int main(int, char**) {
    std::cout << "Hello, world!\n";
}

```

CMakeLists.txt:

``` cmake
cmake_minimum_required(VERSION 3.0.0)
project(dbgtest VERSION 0.1.0)

include(CTest)
enable_testing()

add_executable(dbgtest main.cpp)

set(CPACK_PROJECT_NAME ${PROJECT_NAME})
set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})
include(CPack)

```

launch.json:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "(Windows) Launch",
            "type": "cppvsdbg",
            "request": "launch",
            "program": "${command:cmake.launchTargetPath}",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${fileDirname}",
            "environment": [],
            "console": "internalConsole"
        }
    ]
}
```

**Comment by csorvagep** [comment]
@WardenGnaw, should I provide more info to reproduce this?

**Comment by csorvagep** [comment]
This is still kind of annoying, especially with a graphical interface, this adds an extra window.
